From bb40876fa691983b71fc7231f7dc8ba87f945dd3 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:02:39 -0400
Subject: [usb] ehci: fix type mismatch in check_intr_schedule

Message-id: <1410887463-20674-88-git-send-email-dzickus@redhat.com>
Patchwork-id: 95151
O-Subject: [RHEL7 PATCH 087/591] USB: EHCI: fix type mismatch in check_intr_schedule
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit 5d8f681f3d14880d5b509eb32a910ba90ef28c36
Author: Alan Stern <stern@rowland.harvard.edu>
Date:   Fri Oct 18 11:13:08 2013 -0400

    USB: EHCI: fix type mismatch in check_intr_schedule

    This patch fixes a type mismatch in ehci-hcd caused by commit
    b35c5009bbf6 (USB: EHCI: create per-TT bandwidth tables).  The c_maskp
    parameter in check_intr_schedule() was changed to point to unsigned
    int rather than __hc32, but the prototype declaration wasn't adjusted
    accordingly.

    Signed-off-by: Alan Stern <stern@rowland.harvard.edu>
    Reported-by: kbuild test robot <fengguang.wu@intel.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/host/ehci-sched.c b/drivers/usb/host/ehci-sched.c
index 7ce5c2a..fbeb82e 100644
--- a/drivers/usb/host/ehci-sched.c
+++ b/drivers/usb/host/ehci-sched.c
@@ -794,7 +794,7 @@ static int check_intr_schedule (
  unsigned  frame,
  unsigned  uframe,
  struct ehci_qh  *qh,
- __hc32   *c_maskp,
+ unsigned  *c_maskp,
  struct ehci_tt  *tt
 )
 {
-- 
1.7.1